curvefs/src/client/s3/client_s3_adaptor.cpp

```diff
@@ -197,20 +197,23 @@ CURVEFS_ERROR S3ClientAdaptorImpl::Truncate(Inode *inode, uint64_t size) {
         uint64_t index = offset / chunkSize_;
         uint64_t chunkPos = offset % chunkSize_;
         uint64_t n = 0;
-        uint64_t chunkId;
+        uint64_t beginChunkId;
         FSStatusCode ret;
         uint64_t fsId = inode->fsid();
-        ret = AllocS3ChunkId(fsId, &chunkId);
+        uint64_t chunkIdNum = len / chunkSize_ + 1;
+        ret = AllocS3ChunkId(fsId, chunkIdNum, &beginChunkId);
```

Comment:
Add a test case. A good practice is adding a test case to reproduce issue #1048 firstly, and after your modification, the test should be passed.